src/browser/api_protocol/api_handlers/mesh_middleware.ts

```diff
@@ -148,11 +156,39 @@ function ferryActionMiddleware(msg: MessagePackage, next: () => void) {
     }
 }
 
+function aggregateFromExternalRuntime(msg: MessagePackage, next: (locals?: object) => void) {
+    const { identity, data, ack, nack } = msg;
+    const action = data && data.action;
+    const isAggregateAction = apiMessagesToAggregate[action];
+    //runtimeUuid as part of the identity means the request originated from a different runtime. We do not want to handle it.
+    const isLocalAction = !identity.runtimeUuid;
+
+    try {
+        if (connectionManager.connections.length && isAggregateAction && isLocalAction) {
+            Promise.all(connectionManager.connections.map(runtime => runtime.fin.System.executeOnRemote(identity, data)))
+            .then(data => {
+                const externalRuntimeData = data.reduce((result, runtime) => [...result, ...runtime.data], []);
```

Comment:
My only concern here is that, though right now all the calls return arrays, using `reduce` here over something like `for ... of` means that if ever want to expand this to a call that returns an `object` or `map` in the future this will need to be redone. This does not really bother me too much, but perhaps reduce is less flexible than other.

Reply:
OK - will keep concept in mind going fwd.